js/piwik.js

```diff
@@ -592,6 +592,19 @@ if (typeof window.Matomo !== 'object') {
 
             return matches ? matches[1] : url;
         }
+        function queryStringify(data, filterFn, throwErrorIfNotFilterFn) {
```

Comment:
@geekdenz this method does two things. It validates data plus it builds a query. You'd probably want to split this. Also all uses of this method have `throwErrorIfNotFilterFn=true` meaning the parameter wouldn't even have been needed. Once you split the method things will get simpler as you no longer need to create a default filter function, no longer need this other third parameter etc.